src/Stratis.Bitcoin/P2P/PeerAddress.cs

```diff
@@ -69,6 +69,12 @@ public IPAddress Loopback
         [JsonProperty(PropertyName = "lastConnectionHandshake", NullValueHandling = NullValueHandling.Ignore)]
         public DateTimeOffset? LastConnectionHandshake { get; private set; }
 
+        /// <summary>
+        /// The last time this peer was discovered from.
+        /// </summary>
+        [JsonProperty(PropertyName = "lastDiscoveredFrom", NullValueHandling = NullValueHandling.Ignore)]
+        public DateTime? LastDiscoveredFrom { get; private set; }
```

Comment:
hmm I don't think we need to store this to disk, I suppose this could an in memory parameter that can be stored in the behaviour.

Reply:
>hmm I don't think we need to store this to disk, I suppose this could an in memory parameter that can be stored in the behaviour.

Oh yes, I remember you said that we don't need to persist this to disk. I have fixed this 👍 .

>For example we could have a property called LastGetAddressBroadcastTime

Should I rename LastDiscoveredFrom to LastGetAddressBroadcastTime?

> But more importantly we need to throttle the getaddress that is received not sent, as this is provides a DoS attack possibility.

I can do this in a new PR?

Reply:
@dangershony can you check above questions of mine? Cheers :)

Reply:
Yeah I need to review this better, I will do it soon.